License headers:

kernel/futex.c:
// SPDX-License-Identifier: GPL-2.0-or-later

--- a/kernel/futex/core.c.orig	2021-09-22 14:00:21.491033355 -0700
+++ b/kernel/futex/core.c	2021-09-22 14:02:17.721037185 -0700
@@ -4051,7 +4051,7 @@ SYSCALL_DEFINE6(futex, u32 __user *, uad
 			return -EINVAL;
 
 		t = timespec64_to_ktime(ts);
-		if (cmd == FUTEX_WAIT)
+		if (cmd == FUTEX_WAIT || cmd == FUTEX_WAIT_MULTIPLE)
 			t = ktime_add_safe(ktime_get(), t);
 		else if (cmd != FUTEX_LOCK_PI && !(op & FUTEX_CLOCK_REALTIME))
 			t = timens_ktime_to_host(CLOCK_MONOTONIC, t);
@@ -4254,6 +4254,7 @@ err_unlock:
  */
 struct compat_futex_wait_block {
 	compat_uptr_t	uaddr;
+	__u32 pad;
 	__u32 val;
 	__u32 bitset;
 };
